### PR TITLE
[Fix](statistics)Handle external table in statistics cleaner.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Before, Statistics Cleaner only handles olap db and table. 
External db and tables would be removed without verification. So that external stats could stored no more than 2 days, which is the interval of Stats cleaner thread.
This pr is to add verification for external db and tables.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.38 seconds
 stream load tsv:          535 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162056092 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java [hunk 1]
```diff
@@ -851,7 +851,7 @@ public synchronized EncryptKey getEncryptKey(String keyName) {
         return null;
     }
 
-    public Map<Long, Table> getIdToTable() {
+    public Map<Long, TableIf> getIdToTable() {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java:854]: Add @Override

**Review by morningman — COMMENTED** [review]
Try find a way to add test cases?

**Review by morningman — COMMENTED** [review]
Try find a way to add test cases?

**Comment by Jibing-Li** [comment]
> 

Added

**Comment by Jibing-Li** [comment]
run buildall

**Review by Kikyou1997 — APPROVED** [review]

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -1272,6 +1272,10 @@ clean_stmt ::=
     {:
         RESULT = new CleanQueryStatsStmt(tbl, CleanQueryStatsStmt.Scope.TABLE);
     :}
+    | KW_CLEAN KW_TRASH KW_STATS
```
Comment (morningman) [inline fe/fe-core/src/main/cup/sql_parser.cup:1275]: Need document for this command

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CleanTrashStatsStmt.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import org.apache.doris.common.UserException;
+
+// Clean trash stats
+public class CleanTrashStatsStmt extends DdlStmt {
+
+    public CleanTrashStatsStmt() {}
+
+    @Override
+    public void analyze(Analyzer analyzer) throws UserException {}
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CleanTrashStatsStmt.java:28]: Need to check privilege. I suggest using ADMIN or ROOT

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Comment by Jibing-Li** [comment]
run buildall

**Review by BePPPower — APPROVED** [review]

**Review by morningman — APPROVED** [review]

**Pull request merged**